File: easy/fizzbuzz/fizzbuzz.js
var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        var p = line.split(' ').map(Number);
        fizzbuzz(p[0], p[1], p[2]);
    }
});

function fizzbuzz(x, y, n)
{
    var output = "";
    for (i = 1; i <= n; i++) {
        if (i%x === 0 && i%y === 0) {
            output += "FB";
        } else if (i%x === 0) {
            output += "F";
        } else if (i%y === 0) {
            output += "B";
        } else {
            output += i;
        }
        output += " ";
    }
    console.log(output);
}

File: easy/fizzbuzz/fizzbuzz.rb
File.open(ARGV[0]).each_line do |line|
  x, y, n = line.split.map(&:to_i)

  i = 1
  while i <= n
    if i%x == 0 && i%y == 0
       print 'FB'
     elsif i%x == 0
       print 'F'
     elsif i%y == 0
       print 'B'
     else
       print i
     end
     print " "

     i = i+1
  end
  puts ""

end
